tasks/Task.cpp

```diff
@@ -2,19 +2,57 @@
 
 #include "Task.hpp"
 
+#include <gps_base/BaseTypes.hpp>
+#include <gps_base/rtcm3.hpp>
+
 using namespace gps_ntrip;
+using namespace std;
+
+constexpr int PACKETIZER_BUFFER_SIZE = 4096;
+
+struct RTCMPacketizer : public iodrivers_base::Driver {
+    RTCMPacketizer()
+        : iodrivers_base::Driver(PACKETIZER_BUFFER_SIZE) {}
+
+    int extractPacket(uint8_t const* buffer, size_t size) const override {
+        return gps_base::rtcm3::extractPacket(buffer, size);
+    }
+};
 
 Task::Task(std::string const& name)
-    : TaskBase(name)
+: TaskBase(name)
 {
+    mRTCMPacketizer.reset(new RTCMPacketizer);
+    mRTCMPacketizer->openURI("test://");
+    mRTCMPacketizerStream = dynamic_cast<iodrivers_base::TestStream*>(
```

Comment:
It does not. Just there to avoid having to do the whole dynamic_cast(getMainStream) dance in the updateHook, since it's one of the main object we have to interact with.